File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.go
package main

import (
	"fmt"
	"strconv"
)

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	str := ""
	tmp := ""
	plus := true
	result := 0
	fmt.Scan(&str)

	for i := 0; i <= len(str); i++ {
		if i == len(str) {
			if plus {
				temp, _ := strconv.Atoi(tmp)
				result += temp
			} else {
				temp, _ := strconv.Atoi(tmp)
				result -= temp
			}
			break
		}
		if string(str[i]) == "+" || string(str[i]) == "-" || string(str[i]) == "\n" {
			if plus {
				temp, _ := strconv.Atoi(tmp)
				result += temp
			} else {
				temp, _ := strconv.Atoi(tmp)
				result -= temp
			}
			if string(str[i]) == "-" {
				plus = false
			}
			tmp = ""
			continue
		}
		tmp += string(str[i])
	}
	fmt.Println(result)
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys
sys.setrecursionlimit(2500)


if __name__ == "__main__":

    str_ = sys.stdin.readline()
    tmp = ""
    plus = True
    result = 0

    for i in range(len(str_)):

        if (str_[i] == "+" or str_[i] == "-" or str_[i] == "\n"):
            if plus:
                result += int(tmp)
            else:
                result -= int(tmp)
            if str_[i] == "-":
                plus = False
            tmp = ""
            continue

        tmp += str_[i]

    print(result)
